syllabus/1-Cryptography/zk-proofs/slides.md

```diff
@@ -0,0 +1,446 @@
+
+ $r\times s = N$ 
+
+- Without revealing any other information about $r$ or $s$.
+
+- The statement is $N$ and the witnesses are $r$ and $s$.
+
+
+---
+## ZK Proof properties.
+ - Completeness: If the claim is true, then it *must* pass `verify(statement, proof) == true`
+ - Statistical Knowledge Soundness: If the prover does not know the witness then `verify(statement, proof) == false` with high probability.
+ - Zero Knowledge: the proof reveals nothing about the witness that was not revealed by the statement itself.
+ - The common way of implementing zero knowledge protocol is by means of zk-SNARK.
+
+Notes:
+ 
+ - We prove Zero knowldege by showing a party without any knowledge of the witness, could fool an external party who was not been part of the interaction that they interacted with the prover and got a valid proof form the prover. 
+---
+
+## zk-SNARKs
+
+---
+
+## zk-SNARK
+
+**Z**ero-**K**nowledge **S**uccinct **N**on-interactive **Ar**gument of **K**nowledge
+
+- **Zero knowledge** - the proof reveals nothing about the witness that was not revealed by the statement itself.
+- **Succinct** - the proof is small
+- **Non-interactie** - Does not  require live interaction between prover and the verifier.
+- **Proof of knowledge** - if you can compute correct proofs of a statement, you should be able to compute a witness for it.
+
+---
+
+## What can we show?
+
+- NP relation: `function(statement, witness) -> bool`
+
+  - They want to show that they know it (_a proof of knowledge_)
+
+  - ... Without revealing anything about the witness (_ZK_)
+
+- With a small proof even if the witness is large (_succinctness_)
+
+---
+
+## What can we show?
+
+- There are many SNARK schemes to produce succinct ZK proofs of knowledge (_ZK-SNARKs_) for every NP relation. We concentrate on PLONK in this course.
+
+---
+
+## Applications of zk-SNARKs
+
+---
+
+## ZK Proof Scaling
+
+- A small amount of data, a ZK proof, and execution time can be used to show properties of a much larger dataset which the verifier doesn't need to know. 
+
+- "doesn't want to" know as opposed to "is not supposed to" know.
+
+---
+
+## Scaling via ZK Proofs in Blockchain
+
+- Large amount of data - a blockchain
+- Verifier is e.g. an app on a mobile phone
+
+Notes:
+
+e.g. Mina do a blockchain with a constant size proof (of correctness of execution and consensus) using recursive SNARKs.
+
+---
+
+## Scaling via ZK Proofs in Blockchain
+
+- The verifier is a blockchain: very expensive data and computation costs.
+
+- Layer 2s using ZK rollups
+
+Notes:
+
+Of which Ethereum has many, ZKsync, ZKEVM etc.
+Polkadot already scales better!
+
+---
+
+## Privacy
+
+<pba-flex center>
+
+A user has private data, but we can show<br />publicly that this private data is correctly used.<br />
+An example would a private cryptocurrency:
+
+- Keep who pays who secret
+- Keep amounts secret, <br /> _But show they are positive!_
```

Comment:
This I think is ok though. It is basic enough at this point. We do mention Bitcoin in some of the early hashing lectures as well. I remember Gav doing so aswell.